qiskit/providers/ibmq/api_v2/ibmqversionfinder.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Client for accessing IBM Q's version finder."""
+
+from .session import RetrySession
+from .rest import Auth
+from .rest.version_finder import VersionFinder
+
+
+class IBMQVersionFinder:
+    """Client for programmatic access to the IBM Q API."""
+
+    def __init__(self, auth_url):
```

Comment:
Can you make the constructor accept `url`? The real value of this class is that it will accept any kind of url, and allow deciding if it is an old API url, or a new API (auth or otherwise) url. This would imply that you do not need to create an `Auth` as well - just the minimal amount of objects needed for ending up querying `/version`.